src/test/shared/featureController.test.ts

```diff
@@ -12,27 +12,45 @@ describe('FeatureController', async () => {
     const testConfig = new TestSettingsConfiguration()
     const activeSettingKey = 'activeSetting'
     const inactiveSettingKey = 'inactiveSetting'
+    const activeNotBoolSettingKey = 'activeNotBoolSetting'
+    const inactiveNotBoolSettingKey = 'inactiveNotBoolSetting'
     const missingSettingKey = 'missingSetting'
     const variableSettingKey = 'variableSetting'
     const permanentSettingKey = 'permanentSetting'
     const permanentMissingSettingKey = 'permanentMissingSetting'
     const permanentMissingVariableSettingKey = 'permanentMissingVariableSetting'
-    await testConfig.writeSetting(`toggle.${activeSettingKey}`, true)
-    await testConfig.writeSetting(`toggle.${inactiveSettingKey}`, false)
-    await testConfig.writeSetting(`toggle.${variableSettingKey}`, false)
-    await testConfig.writeSetting(`toggle.${permanentSettingKey}`, true)
+    let features: FeatureController
 
-    const features = new FeatureController(testConfig, [
-        permanentSettingKey,
-        permanentMissingSettingKey,
-        permanentMissingVariableSettingKey
-    ])
+    beforeEach(async () => {
+        await testConfig.writeSetting(`toggle.${activeSettingKey}`, true)
```

Comment:
Can you instantiate testConfig  in here too so that each test works with a fresh copy?